src/inmanta/data/model.py

```diff
@@ -234,7 +235,7 @@ def check_serializable(cls, v: Optional[Any]) -> Optional[Any]:
         return v
 
 
-EnvSettingType = Union[StrictNonIntBool, int, float, str, Dict[str, Union[str, int, StrictNonIntBool]]]
+EnvSettingType = Union[StrictNonIntBool, StrictInt, StrictFloat, StrictStr, Dict[str, Union[str, int, StrictNonIntBool]]]
```

Comment:
Without strict validation, e.g. `"86400"` would pass validation as an `int` (which comes before `str` in the list) and get type coerced into `86400`.